src/App.jsx

```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABET_SORT_VALUE = 'alphabet';
+const LENGTH_SORT_VALUE = 'length';
+const DEFAULT_SORT_VALUE = '';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case ALPHABET_SORT_VALUE:
+          return good1.localeCompare(good2);
+
+        case LENGTH_SORT_VALUE:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(DEFAULT_SORT_VALUE);
+  const [isReversed, setIsReversed] = useState(false);
+  const reverseButton = (sortField || isReversed);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const initialGoods = getPreparedGoods(
+    goodsFromServer,
+    {
+      sortField,
+      isReversed,
+    },
+  );
+
+  const handleButtonReset = () => {
+    setSortField(DEFAULT_SORT_VALUE);
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(`button`, `is-info`, {
+            'is-light': sortField !== ALPHABET_SORT_VALUE,
+          })}
```

Comment:
```suggestion
          className={cn(
            `button`,
            `is-info`, 
            { 'is-light': sortField !== ALPHABET_SORT_VALUE },
          )}
```
Try write your code more readable